### ISSUE TITLE
AD0001 - Interpolated string with range operator

### ISSUE THREAD

**Issue by AdrianoAE** [dotnet/roslyn#54364]
**Version Used**: 
.Net 5
Visual Studio 16.10.1

**Steps to Reproduce**:
```csharp
// This one sometimes did throw the error and others didn't for some reason
string test = "potato";
_ = $"{test[..]}";
```
or
```csharp
var test = "potato".AsSpan();
_ = $"{test[..].ToString()}";
```

**Expected Behavior**:
No errors

**Actual Behavior**:
The problem seems to be with the range operator in interpolated strings, even after removing those lines the error won't go away without a restart and it was very difficult to understand where the problem was because most of the time it only shows this error
```
Severity	Code	Description	Project	File	Line
Error	AD0001	Analyzer 'Microsoft.CodeAnalysis.CSharp.SimplifyInterpolation.CSharpSimplifyInterpolationDiagnosticAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.	Test	C:\...\Test\CSC	1
```

On code write/paste it shows the stack trace

```
Severity	Code	Description	Project	File	Line
Error	AD0001	Analyzer 'Microsoft.CodeAnalysis.CSharp.SimplifyInterpolation.CSharpSimplifyInterpolationDiagnosticAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
Exception occurred with following context:
Compilation: Test
IOperation: Interpolation
SyntaxTree: C:\...\Test.cs
SyntaxNode: {test[..].ToString()} [InterpolationSyntax]@[2611..2632) (80,10)-(80,31)

System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.CodeAnalysis.SimplifyInterpolation.AbstractSimplifyInterpolationHelpers.UnwrapFormatString(IVirtualCharService virtualCharService, ISyntaxFacts syntaxFacts, IOperation expression, IOperation& unwrapped, String& formatString, List`1 unnecessarySpans)
   at Microsoft.CodeAnalysis.SimplifyInterpolation.AbstractSimplifyInterpolationHelpers.UnwrapInterpolation[TInterpolationSyntax,TExpressionSyntax](IVirtualCharService virtualCharService, ISyntaxFacts syntaxFacts, IInterpolationOperation interpolation, TExpressionSyntax& unwrapped, TExpressionSyntax& alignment, Boolean& negate, String& formatString, ImmutableArray`1& unnecessaryLocations)
   at Microsoft.CodeAnalysis.SimplifyInterpolation.AbstractSimplifyInterpolationDiagnosticAnalyzer`2.AnalyzeInterpolation(OperationAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteOperationAction>b__63_0(ValueTuple`2 data)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
-----
```
